### PR TITLE
Ignore frozen shared cache file during data folder upgrades

### LINKED ISSUE #85603: Frozen cache files are considered as unknown data files preventing node startup during upgrades
During upgrades from 7.x to 8.x if a node was holding a frozen cache file, the system considers that as an unknown file and prevents starting the node:
```
java.lang.IllegalStateException: unexpected file/folder encountered during data folder upgrade: /app/data/nodes/0/shared_snapshot_cache
	at org.elasticsearch.env.NodeEnvironment.upgradeLegacyNodeFolders(NodeEnvironment.java:443) ~[elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.env.NodeEnvironment.<init>(NodeEnvironment.java:290) ~[elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.node.Node.<init>(Node.java:388) ~[elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.node.Node.<init>(Node.java:277) ~[elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:234) ~[elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:234) ~[elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:358) [elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:166) [elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:157) [elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:77) [elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:112) [elasticsearch-cli-8.0.1.jar:8.0.1]
	at org.elasticsearch.cli.Command.main(Command.java:77) [elasticsearch-cli-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:122) [elasticsearch-8.0.1.jar:8.0.1]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:80) [elasticsearch-8.0.1.jar:8.0.1]
```
These files are expected files and should be ignored/accepted during the data folder upgrade.

### PR DESCRIPTION
**Pull request by fcofdez**
Closes #85603

### CODE REVIEW

**Comment by fcofdez** [comment]
I've added `xpack.searchable.snapshot.shared_cache.region_size` in some other spots where it was using small caches to see if more bugs arise. I'll tackle the validation in a separate PR.

**Review by DaveCTurner — APPROVED** [review]
LGTM

@elasticmachine please run elasticsearch-ci/packaging-tests-windows-sample

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/packaging-tests-windows-sample

File: server/src/internalClusterTest/java/org/elasticsearch/env/NodeEnvironmentIT.java [hunk 1]
```diff
@@ -216,6 +217,13 @@ public void testUpgradeDataFolder() throws IOException, InterruptedException {
             }
         }
 
+        // simulate a frozen node with a shared cache file
+        if (randomBoolean()) {
```
Comment (tlrx) [inline server/src/internalClusterTest/java/org/elasticsearch/env/NodeEnvironmentIT.java:221]:
```suggestion
        if (rarely()) {
```

**Review by tlrx — APPROVED** [review]
LGTM, thanks for fixing this Francisco!

**Comment by fcofdez** [comment]
@elasticmachine update branch

**Pull request merged**